cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +105,300 @@ def unset_default_org(self):
+        service_path = Path(
+            f"{self.global_config_dir}/services/{service_type}/{alias}.service"
+        )
+        with open(service_path, "wb") as f:
+            f.write(encrypted)
+
+    def _init_default_services(self) -> None:
+        """Init self._default_services on the keychain so that
+        calls to get_service() that do not pass an alias can
+        return the default service for the given type"""
+        global_default_services = Path(
+            f"{self.global_config_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        self._set_default_services_for_dir(global_default_services)
+
+        project_default_services = Path(
+            f"{self.project_local_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        # project defaults will overwrite global defaults
+        self._set_default_services_for_dir(project_default_services)
+
+    def _set_default_services_for_dir(self, dir_path: Path) -> None:
+        if dir_path.is_file():
+            with open(dir_path, "r") as f:
+                default_services = json.loads(f.read())
+
+            for s_type, alias in default_services.items():
+                self._default_services[s_type] = alias
+
+    def _save_default_services(self, project: bool = False) -> None:
+        """Write out the contents of self._default_services to the
+        DEFAULT_SERVICES.json file based on the provided scope"""
+        dir_path = (
+            Path(self.project_local_dir) if project else Path(self.global_config_dir)
+        )
+        with open(dir_path / DEFAULT_SERVICES_FILENAME, "w") as f:
+            f.write(json.dumps(self._default_services))
+
+    def _create_default_service_files(self) -> None:
+        """
+        Generate the DEFAULT_SERVICES.json files at global and project scopes.
+
+        @param local_proj_path: should be the local_project_path for the project
+        """
+        global_default_service_file = Path(
+            f"{self.global_config_dir}/{DEFAULT_SERVICES_FILENAME}"
+        )
+        if global_default_service_file.is_file():
+            return
+
+        self._write_default_services_for_dir(self.global_config_dir)
+        for local_proj_dir in self._iter_local_project_dirs():
+            self._write_default_services_for_dir(local_proj_dir)
+
+    def _write_default_services_for_dir(self, dir_path: str) -> None:
+        """Look through the given dir and set all .service files
+        present as the defaults for their given types, and write these
+        out to a DEFAULT_SERVICES.json file in the directory. This occurs
+        once before .service files are migrated to the appropriate
+        services/ sub-directory, so we set the default to the alias
+        that will be assigned during migration.
+
+        @param dir_path: the directory to look through and write
+        the DEFAULT_SERVICES.json file in.
+        """
+        if dir_path is None:
+            return  # pragma: no cover
```

Comment:
Can this actually happen?

Reply:
Just tested and looks like we can remove this.